### DISCUSSION TITLE
The edge are overlapping over each other when source and target are same....I'm creating a condition node which has true and false edges and these both edges are converging on a single end node

### DISCUSSION THREAD

**Discussion by Kritica1** [xyflow/xyflow#3676] [Q&A]
``The edges are over lapping 
<img width="200" alt="image" src="https://github.com/xyflow/xyflow/assets/150110042/09fe309c-5b8f-469a-87e0-4640c62598d8">
It has two edges one true and other false, but it's overlapping one over another since the source and target both are same, please suggest some solution.
<img width="425" alt="image" src="https://github.com/xyflow/xyflow/assets/150110042/85b0005b-ec36-4d9c-b40e-7cca26d54cf0">
When I first add those edges it should also come like this curvy
<img width="425" alt="image" src="https://github.com/xyflow/xyflow/assets/150110042/b57ed844-3fb7-48c2-ad76-bc3089a4f81a">

```
import React from 'react';
import { BaseEdge, EdgeLabelRenderer, getBezierPath } from 'reactflow';

const CustomEdge = ({
  id,
  sourceX,
  sourceY,
  targetX,
  targetY,
  sourcePosition,
  targetPosition,
  style = {},
  markerEnd,
  data,
  label,
  ...props
}: any) => {
  const [edgePath, labelX, labelY] = getBezierPath({
    sourceX,
    sourceY,
    sourcePosition,
    targetX,
    targetY,
    targetPosition,
  });

  return (
    <>
      <BaseEdge
        path={edgePath}
        markerEnd={markerEnd}
        style={style}
        {...props}
        data={data}
      />
      <EdgeLabelRenderer>
        <div
          style={{
            position: 'absolute',
            transform: `translate(-50%, -50%) translate(${labelX}px,${labelY}px)`,
            fontSize: 12,
            pointerEvents: 'all',
            color: 'black',
            display: 'flex',
            alignItems: 'center',
          }}
          className={
            label
              ? `${classes.customEdgeLabel} ${
                  label === 'True' ? classes.edgeTrue : classes.edgeFalse
                }`
              : ''
          }
        >
          <div>
            {label !== 'True' && label !== 'False' && (
              <button className={classes.edgebtn}>
                <AddCircle24Regular className={classes.customEdge} />
              </button>
            )}
          </div>
          {label && <div>{label}</div>}
        </div>
      </EdgeLabelRenderer>
    </>
  );
};
export default CustomEdge;
```

This is myCustomEdge

**Comment by moklick** [comment]
which edges are overlapping? All images look good to me.

**Reply by Kritica1** [reply]
Hi @moklick , 
<img width="245" alt="image" src="https://github.com/xyflow/xyflow/assets/150110042/ed261f05-a6d1-4f27-a6e4-7ee5c55bf8b6">
So, here I have added two edges, the target for both the edges is same(END). I have also added label for both True, false but that's overlapping...how can we make it look like two edges.
I hope my doubt is clear, 
thanks for the response

**Reply by moklick** [reply]
This is only solvable with a custom edge. You would need to create an edge that can handle the case that there are two edges going from the same source to the same target. In that case, you want to draw your special edge. Check out the BiDirectionalEdge here, it's a quire similar use case: https://reactflow.dev/examples/edges/custom-edges

**Reply by Kritica1** [reply]
Hi, thanks for the response, but I'm not able to correctly calculate the path, a quick example would really be very helpful
@moklick

**Reply by djibomar** [reply]
I have a similar use case (if/else node) and  facing the same issue. Have you made any progress on this ? @moklick  do you have an example you can share ?

**Reply by Kritica1** [reply]
https://codesandbox.io/p/sandbox/gallant-poincare-f7q99x?file=%2FBiDirectionalEdge.tsx
